pilz_control/test/unittest_traj_mode_state_machine.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2020 Pilz GmbH & Co. KG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <pilz_control/traj_mode_manager.h>
+
+namespace pilz_joint_trajectory_controller
+{
+
+TEST(TrajModeStateMachineTest, testIsTransitionValid)
+{
+  for (unsigned int i = 0; i < NUM_MODES; ++i)
+  {
+    auto mode = getMode(i);
+    EXPECT_TRUE(isTransitionValid(mode, i));
+  }
+}
+
+TEST(TrajModeStateMachineTest, testNextTransitionInHoldMode)
+{
+  TrajProcessingMode mode{TrajProcessingMode::hold};
+  bool found_index{false};
+  for (unsigned int i = 0; i < NUM_MODES; ++i)
+  {
+    if (isTransitionValid(mode, i))
```

Comment:
Actually, I find the function `isTransitionValid` misleading. A transition is defined by two states (modes). I would suggest to refactor it such that it takes the current index instead of the new index. What do you think?

The search function for the tests makes sense.

Reply:
>  I would suggest to refactor it such that it takes the current index instead of the new index.

That makes no sense to me. I want to test that my intended transition from one mode to another is valid. Therefore, I have to know the current mode is and what the requested new mode is. Using the current_index and the mode belonging to the current_index, I'm missing the information which transition is requested and, therefore, the check makes no sense...

Reply:
> Therefore, I have to know the current mode and the requested new mode.

That is my point. Currently the function takes the requested new mode plus the new mode index.

Reply:
I decided to change the code completely. I simplified the state machine handling drastically. Now, only the function `isTransitionValid()` exists which takes the current mode and the requested mode and states if a transition exists or not. Hopefully, this is what you wanted.